In [9]:
import os

import numpy as np
import pandas as pd

import dicom

In [2]:
data_dir = '/labs/gevaertlab/users/hackhack/RTOG/scratch'

# get the raw names of the folders to classify
all_files = []
all_files_path = []
for study in os.listdir(data_dir):  
    if 'Studies' in study:
        study_path = os.path.join(data_dir, study)
        for filename in os.listdir(study_path):
            filename_path = os.path.join(study_path, filename)
            # exclude folders with no dicoms
            if len(os.listdir(filename_path)) > 0:
                full_name = study + '/' + filename
                all_files.append(full_name)
                all_files_path.append(filename_path)

In [6]:
# to play with an example
ex = all_files_path[11]
ex_dicoms = os.listdir(ex)
ex_dicoms = [dic for dic in ex_dicoms if dic[-3:] == 'dcm']
ex_dicoms = [os.path.join(ex, dic) for dic in ex_dicoms]
ex_dic = ex_dicoms[10]

In [10]:
print(ex_dic)
pat = dicom.read_file(ex_dic)

pat

/labs/gevaertlab/users/hackhack/RTOG/scratch/0101-01__Studies/727^1027^825_727_MR_0101-01-01_103407_BRAIN.W.WO.CM.MRI_AXIAL.T2.FLAIR_n26__00000/1.2.840.113619.2.244.3596.12002590.21165.1299586853.123.dcm


(0008, 0005) Specific Character Set              CS: 'ISO_IR 100'
(0008, 0008) Image Type                          CS: ['ORIGINAL', 'PRIMARY', 'OTHER']
(0008, 0016) SOP Class UID                       UI: MR Image Storage
(0008, 0018) SOP Instance UID                    UI: 1.2.840.113619.2.244.3596.12002590.21165.1299586853.123
(0008, 0020) Study Date                          DA: '010101'
(0008, 0021) Series Date                         DA: '20110308'
(0008, 0022) Acquisition Date                    DA: '20110308'
(0008, 0023) Content Date                        DA: '20110308'
(0008, 0030) Study Time                          TM: '103407'
(0008, 0031) Series Time                         TM: '104420'
(0008, 0032) Acquisition Time                    TM: '104420'
(0008, 0033) Content Time                        TM: '104420'
(0008, 0050) Accession Number                    SH: '0'
(0008, 0060) Modality                            CS: 'MR'
(0008, 0070) Manufacturer                        LO:

In [6]:
# check that all dicoms have same description
count = 0
weird = []

for ex in all_files_path:
    ex_dicoms = os.listdir(ex)
    ex_dicoms = [dic for dic in ex_dicoms if dic[-3:] == 'dcm']
    ex_dicoms = [os.path.join(ex, dic) for dic in ex_dicoms]
    descriptions = set()
    for ex_dic in ex_dicoms:
        pat = dicom.read_file(ex_dic)
        try:
            descriptions.add(pat.SeriesDescription)
        except AttributeError:
            continue
    if len(descriptions) != 1:
        weird.append(ex)
        count += 1

useful fields:
- Acquisition Date (e.g. 20110308)
- Manufacturer (e.g. GE MEDICAL SYSTEMS)
- Institution Name (e.g. Anonymous Hospital149360)
- SeriesDescription (e.g. AXIAL T2 FLAIR)
- PatientName (e.g. 727^1027^825^^)
- Clinical Trial Sponsor Name (e.g. American_College_of_Radiology)
- Clinical Trial Site Name (e.g. Hartford Hospital)
- Slice Thickness (e.g. 5)
- Spacing between Slices (e.g. 7)
- Flip angle (e.g. 90)
- Series Contrast (e.g. 0)
- Stack ID (e.g. 1)
- In-Stack Position Number (e.g. 3) -- very important to stack in order!!!
- Image dimension - X (e.g. 256)
- Image dimension - Y (e.g. 192)
- Rows (e.g. 512)
- Columns (e.g. 512)
- Pixel Spacing (e.g. ['0.4688', '0.4688'])
- Smallest Image Pixel Value (e.g. 0)
- Largest Image Pixel Value (e.g. 1094)

### inspect rtog data vs brats data